sdk/src/test/java/io/dapr/client/DaprClientHttpTest.java

```diff
@@ -189,13 +190,80 @@ public void invokeServiceIllegalArgumentException() {
     });
   }
 
+  @Test
+  public void invokeServiceDaprError() {
+    mockInterceptor.addRule()
+        .post("http://127.0.0.1:3000/v1.0/invoke/myapp/method/mymethod")
+        .respond(500,
+            ResponseBody.create(
+                "{ \"errorCode\": \"MYCODE\", \"message\": \"My Message\"}",
+                MediaTypes.MEDIATYPE_JSON));
+
+    DaprException exception = assertThrows(DaprException.class, () -> {
+      daprClientHttp.invokeMethod("myapp", "mymethod", "anything", HttpExtension.POST).block();
+    });
+
+    assertEquals("MYCODE", exception.getErrorCode());
+    assertEquals("MYCODE: My Message", exception.getMessage());
+  }
+
+  @Test
+  public void invokeServiceDaprErrorFromGRPC() {
+    mockInterceptor.addRule()
+        .post("http://127.0.0.1:3000/v1.0/invoke/myapp/method/mymethod")
+        .respond(500,
```

Comment:
please have 403 to match gRPC code.

Reply:
Actually, 500 has nothing to do with the http gRPC status code. We do not match that at all.